src/main/java/com/undertheriver/sgsg/foler/service/FolderService.java

```diff
@@ -0,0 +1,58 @@
+package com.undertheriver.sgsg.foler.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.undertheriver.sgsg.common.dto.CurrentUser;
+import com.undertheriver.sgsg.foler.domain.Folder;
+import com.undertheriver.sgsg.foler.domain.dto.FolderDto;
+import com.undertheriver.sgsg.foler.repository.FolderRepository;
+import com.undertheriver.sgsg.user.domain.User;
+
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@Service
+public class FolderService {
+	private final FolderRepository folderRepository;
+
+	private static final Integer FOLDER_INDEX_LIMIT = 20;
+
+	@Transactional
+	public Long save(Long userId, FolderDto.CreateFolderReq req) {
+		if (isFoldersExistsMoreThan20(userId)) {
+			throw new IndexOutOfBoundsException("폴더는 최대 20개까지 생성할 수 있습니다!");
+		}
+		return folderRepository.save(req.toEntity()).getId();
+	}
```

Comment:
```suggestion
	public Long save(Long userId, FolderDto.CreateFolderReq req) {
		if (isFoldersExistsMoreThan(userId, FOLDER_INDEX_LIMIT)) {
			throw new IndexOutOfBoundsException(String.format("폴더는 최대 %d개까지 생성할 수 있습니다!", FOLDER_INDEX_LIMIT));
		}
		return folderRepository.save(req.toEntity()).getId();
	}
```

쿼리에 20이라는 수가 고정으로들어가니까, 변화에 유연하지 않을 것 같습니다. 

FOLDER_INDEX_LIMIT가 변경되면 쿼리도 변경되어야할 것 같아요. 쿼리인자에 `FOLDER_INDEX_LIMIT`를 입력받도록하는건 어떨까요?

또 FOLDER_INDEX_LIMIT이 변수도 application.yml에서 관리해 DI하면, FOLDER_INDEX_LIMIT가 바뀔때 소스코드변경할 필요가없을 것 같습니다. 변화에 유연할 것 같아요

Reply:
그렇네요 기껏 상수 선언 해놓고 제대로 활용을 하지를 않았네요. 말씀주신대로 application.yml에 반영했습니다.

테스트 코드용 application.yml에도 반영을 해야 해서 상수 전용 application.-*yml을  만들고 include 시키고 싶은데 이름 짓기가 좀 어렵네요

application-const.yml 로 해볼까 생각했는데 앞으로 모든 상수들이 여기에 들어 갈 것 같지는 않아서 적합해 보이지는 않고 .. 
흠..좋은 의견이 있을까요?

Reply:
그렇군요
그럼 일단 application-auth.yml로 변경하고 진행해보죠. 해당 파일은 공개해도 상관없을듯하네요.
혹은 application.yml에 넣어도 될 것 같구요

Reply:
넵 일단은 application.yml 에 넣어서 푸시해놨습니다